src/globus_cli/commands/flows/delete.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import annotations
+
+import uuid
+
+from globus_cli.login_manager import LoginManager
+from globus_cli.parsing import command, flow_id_arg
+from globus_cli.termio import Field, TextMode, display, formatters
+
+
+@command("delete", short_help="Delete a flow")
+@flow_id_arg
+@LoginManager.requires_login(LoginManager.FLOWS_RS)
+def delete_command(login_manager: LoginManager, flow_id: uuid.UUID):
+    """
+    Delete a flow
+    """
+    flows_client = login_manager.get_flows_client()
+
+    fields = [
+        Field("Deleted", "DELETED", formatter=formatters.Bool),
+        Field("Flow ID", "id"),
+        Field("Title", "title"),
+        Field(
+            "Owner",
+            "flow_owner",
+            formatter=formatters.auth.PrincipalURNFormatter(
+                login_manager.get_auth_client()
```

Comment:
The call to the login manager here doesn't have a side-effect (such as some network I/O) that the `flows_client` relies on, does it?

Reply:
No, it's just grabbing credentials out of the on-disk sqlite storage and using them to build a client object.

If you look at how the `login_manager` gets passed, you'll see that it's through a `requires_login` decorator -- that decorator's job is to ensure the user is properly credentialed. If the user doesn't have creds for the clients listed there, the decorator hard-fails the command with a message instructing them to login.